src/components/TableSettings/__stories__/stories/TableSettingsColumnStory.tsx

```diff
@@ -97,17 +97,27 @@ const columns: ColumnDef<Item>[] = [
 ];
 
 export const TableSettingsColumnStory = (options: TableSettingsOptions) => {
+    const [rowSelection, setRowSelection] = React.useState<RowSelectionState>({});
     const settingsColumn = getSettingsColumn<Item>(SETTINGS_COLUMN_ID, options);
     const [columnPinning, setColumnPinning] = React.useState<ColumnPinningState>({
-        left: [],
+        left: ['_select'],
         right: [SETTINGS_COLUMN_ID],
     });
+
+    const columns = React.useMemo(
+        () => [selectionColumn as ColumnDef<Item>, ...tableColumns, settingsColumn],
+        [settingsColumn],
```

Comment:
Move `getSettingsColumn` to this `useMemo`:
```
export const TableSettingsColumnStory = ({sortable, filterable}: TableSettingsOptions) => {
    // ...

    const columns = React.useMemo(
        () => [
            selectionColumn as ColumnDef<Item>,
            ...tableColumns,
            getSettingsColumn<Item>(SETTINGS_COLUMN_ID, {sortable, filterable}),
        ],
        [sortable, filterable],
    );

    // ...
}
```

Reply:
Moved